app/scripts/controllers/MainCtrl.js

```diff
@@ -23,14 +26,17 @@
 			$scope.nameOfRoom = room.room;
 		}
 
-		$scope.removeUser = function() {
-			$cookies.remove('blocChatCurrentUser');
-			console.log("cookie removed " + $cookies.getAll());
-			$location.path('/set-username');
+		$scope.sendMessage = function(messageContent, roomId) {
+			var roomId = $scope.activeRoom.$id;
+			$scope.send(messageContent, roomId);
```

Comment:
As noted above, change this to `Message.send`.

For `messageContent` the problem you have with clearing is that `ng-include` creates a new child `$scope`. I would suggest the following to work _with_ that rather than try to work around it:

- at the top of the controller define `$scope.messageContent = null` to act as a model for the message input.

- in `send-message.html` change the `input` `ng-model` to:

```
ng-model="$parent.messageContent"
```

This is now the same model as `$scope.messageContent` you added above.

- access the model directly now, instead of passing it as an argument to the method:

```
Message.send($scope.messageContent, roomId);
$scope.messageContent = '';
```